src/test/java/Problem3Test.java

```diff
@@ -146,9 +172,12 @@ private static void inOrderTraverse(TreeNode<Integer> node, List<Integer> result
         //    N   N
         // homework
         // what problem can you see for insertInBst from this test case?
-        // answer:
+        // answer: The issue with this test case is that the expected test cases do not account for
+        // the previous values being added to the tree, for instance the second test case adds 3 to the list
+        // but it also counts on 2 being already in the tree, which it won't be.
```

Comment:
this test case didn't fail. the issue here is a binary search tree will end up like a single linked list because all values added will go to the right tree